skimage/metrics/_adapted_rand_error.py

```diff
@@ -0,0 +1,78 @@
+from .._shared.testing import assert_shape_equal
+from ._contingency_table import contingency_table
+
+__all__ = ['adapted_rand_error']
+
+
+def adapted_rand_error(im_true=None, im_test=None, *, table=None,
+                       ignore_labels=None, normalize=False):
+    r"""Compute Adapted Rand error as defined by the SNEMI3D contest. [1]_
+
+    Parameters
+    ----------
+    im_true : ndarray of int
+        Ground-truth label image, same shape as im_test.
+    im_test : ndarray of int
+        Test image.
+    table : scipy.sparse array in crs format, optional
+        A contingency table built with skimage.evaluate.contingency_table.
+        If None, it will be computed on the fly.
+    ignore_labels : list of int, optional
+        Labels to ignore. Any part of the true image labeled with any of these
+        values will not be counted in the score.
+    normalize : bool, optional
+        If True, normalizes contigency table by the number of pixels of
+        each value.
+
+    Returns
+    -------
+    are : float
+        The adapted Rand error; equal to $1 - \frac{2pr}{p + r}$,
+        where $p$ and $r$ are the precision and recall described below.
+    prec : float
+        The adapted Rand precision: this is the number of pairs of pixels that
+        have the same label in the test label image *and* in the true image,
+        divided by the number in the test image.
+    rec : float
+        The adapted Rand recall: this is the number of pairs of pixels that
+        have the same label in the test label image *and* in the true image,
+        divided by the number in the true image.
+
+    Notes
+    -----
+    Pixels with label 0 in the true segmentation are ignored in the score.
+
+    References
+    ----------
+    .. [1] Arganda-Carreras I, Turaga SC, Berger DR, et al. (2015)
+           Crowdsourcing the creation of image segmentation algorithms
+           for connectomics. Front. Neuroanat. 9:142.
+           :DOI:`10.3389/fnana.2015.00142`
+    """
+    assert_shape_equal(im_true, im_test)
```

Comment:
This comment follows another inline comment I made where this function is defined.

I don't like calling this function assert_something, because it does not call assert. And this is important. Assert statements are removed if python is called with an optimization flag.

So, assert_shape_equal is not written in the right file. I'm not sure where it can fit the best and I'm not certain that we want this sort of functions in scikit-image. @scikit-image/core Thoughts?